File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        i, j = 0, n - 1
        ans = [0, 0]
        while i < j:
            sums = nums[i] + nums[j]
            if sums == target: 
                ans[0] = i + 1
                ans[1] = j + 1
                return ans
            elif sums > target: 
                j -= 1
            else:
                i += 1
        return ans

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        int i = 0, j = n - 1;
        int[] ans = new int[2];
        while(i < j) {
            int sum = nums[i] + nums[j];
            if(sum == target) { ans[0] = i + 1; ans[1] = j + 1; return ans; }
            else if(sum > target) --j;
            else ++i;
        }
        return ans;
    }
}